src/sqlBible/src/SqlBible.ts

```diff
@@ -0,0 +1,292 @@
+        return new BibleCrossReference(this.getNormalizedReference(refRange));
+    }
+
+    async generateBookMetadata(book: BibleBook) {
+        await this.dbReady;
+        const metaData = await getManager()
+            .createQueryBuilder(BiblePhrase, 'phrase')
+            .addSelect('COUNT(DISTINCT phrase.versionVerseNum)', 'numVerses')
+            .where({
+                id: Between(
+                    generateNormalizedRefId(
+                        {
+                            bookOsisId: book.osisId,
+                            normalizedChapterNum: 1,
+                            normalizedVerseNum: 1,
+                            versionId: book.versionId
+                        },
+                        5
+                    ),
+                    generateNormalizedRefId(
+                        {
+                            bookOsisId: book.osisId,
+                            normalizedChapterNum: 999,
+                            normalizedVerseNum: 999,
+                            versionId: book.versionId
+                        },
+                        5
+                    )
+                )
+            })
+            .orderBy('phrase.versionChapterNum')
+            .groupBy('phrase.versionChapterNum')
+            .getRawMany();
+        book.setChaptersMeta(metaData.map(chapterMetaDb => chapterMetaDb.numVerses));
+        return getManager().save(book);
+        console.log(metaData);
+    }
+
+    async getBookForVersionReference(reference: IBibleReferenceVersion) {
+        await this.dbReady;
+        return getManager().findOne(BibleBook, {
+            where: {
+                versionId: reference.versionId,
+                osisId: reference.bookOsisId
+            }
+        });
+    }
+
+    async getBooksForVersion(versionId: number) {
+        await this.dbReady;
+        return getManager().find(BibleBook, {
+            where: {
+                versionId
+            },
+            order: {
+                number: 'ASC'
+            }
+        });
+    }
+
+    async getNextPhraseIdForNormalizedVerseNum(
+        reference: IBibleReferenceNormalized,
+        versionId: number
+    ): Promise<number> {
+        await this.dbReady;
+        const lastPhrase = await getManager().find(BiblePhrase, {
+            where: {
+                id: Between(
+                    generateNormalizedRefId({ ...reference, versionId }, 5),
+                    generateNormalizedRefId(
+                        {
+                            ...reference,
+                            versionId,
+                            phraseNum: 99
+                        },
+                        5
+                    )
+                )
+            },
+            order: {
+                id: 'DESC'
+            },
+            take: 1,
+            select: ['id']
+        });
+        return lastPhrase.length ? parsePhraseId(lastPhrase[0].id).phraseNum! + 1 : 1;
```

Comment:
We should be throwing an error if `lastPhrase.length` is zero, right? because it basically means that a phrase wasn't found

Reply:
this method does not intend to find a phrase (even then i think we shouldn't throw an error since not finding a phrase it not something totally unexpected). anyway, it's totally expected that the phrase isn't there - if not that means we can start numbering phrases at 1.